x/emissions/keeper/actor_utils/losses.go

```diff
@@ -98,77 +98,58 @@ func CloseReputerNonce(
+		if err != nil {
+			continue
+		}
+		if stake.LT(params.RequiredMinimumStake) {
+			continue
+		}
 
-			// Examine forecast elements to verify that they're for registered inferers in the current set.
-			// A check of their registration and other filters have already been applied when their inferences were inserted.
-			// We keep what we can, ignoring the reputer and their contribution (losses) entirely
-			// if they're left with no valid losses.
+		// Examine forecast elements to verify that they're for registered inferers in the current set.
+		// A check of their registration and other filters have already been applied when their inferences were inserted.
+		// We keep what we can, ignoring the reputer and their contribution (losses) entirely
+		// if they're left with no valid losses.
 
-			filteredBundle, err := filterUnacceptedWorkersFromReputerValueBundle(k, ctx, topicId, *bundle.ValueBundle.ReputerRequestNonce, bundle)
-			if err != nil {
-				continue
-			}
+		filteredBundle, err := filterUnacceptedWorkersFromReputerValueBundle(k, ctx, topicId, *bundle.ValueBundle.ReputerRequestNonce, bundle)
+		if err != nil {
+			continue
+		}
 
-			/// If we do PoX-like anti-sybil procedure, would go here
+		/// If we do PoX-like anti-sybil procedure, would go here
 
-			/// Filtering done now, now write what we must for inclusion
+		/// Filtering done now, now write what we must for inclusion
 
-			// Get the latest score for each reputer
-			latestScore, err := k.GetLatestReputerScore(ctx, bundle.ValueBundle.TopicId, reputer)
-			if err != nil {
-				continue
-			}
-			latestReputerScores[bundle.ValueBundle.Reputer] = latestScore
-			lossBundlesByReputer[bundle.ValueBundle.Reputer] = filteredBundle
+		if err != nil {
+			continue
 		}
-	}
+		lossBundlesByReputer = append(lossBundlesByReputer, filteredBundle)
 
-	// If we pseudo-random sample from the non-sybil set of reputers, we would do it here
-	topReputers := FindTopNByScoreDesc(params.MaxTopReputersToReward, latestReputerScores, nonce.BlockHeight)
-
-	// Check that the reputer in the payload is a top reputer among those who have submitted losses
-	stakesByReputer := make(map[string]cosmosMath.Int)
-	lossBundlesFromTopReputers := make([]*types.ReputerValueBundle, 0)
-	for _, reputer := range topReputers {
-		stake, err := k.GetStakeReputerAuthority(ctx, topicId, reputer)
+		stake, err = k.GetStakeReputerAuthority(ctx, topicId, reputer)
 		if err != nil {
 			continue
 		}
-
-		lossBundlesFromTopReputers = append(lossBundlesFromTopReputers, lossBundlesByReputer[reputer])
-		stakesByReputer[reputer] = stake
+		stakesByReputer[bundle.ValueBundle.Reputer] = stake
 	}
+
 	// sort by reputer score descending
-	sort.Slice(lossBundlesFromTopReputers, func(i, j int) bool {
-		return lossBundlesFromTopReputers[i].ValueBundle.Reputer < lossBundlesFromTopReputers[j].ValueBundle.Reputer
+	sort.Slice(lossBundlesByReputer, func(i, j int) bool {
```

Comment:
not sure you have to sort if you were already skimming the top as the bundles were being submitted...

Reply:
When submitting data, only the append function is performed. Therefore, I think sorting should be done at close time.